internal/pkg/utils/error.go

```diff
@@ -19,33 +20,25 @@ type MultiError struct {
 
 type NestedError struct {
 	*MultiError
-	msg string
+	Msg string
 }
 
-func (e *NestedError) Error() (out string) {
-	l := e.MultiError.Len()
+func NewMultiError() *MultiError {
+	e := &MultiError{multiError: &multierror.Error{}, lock: &sync.Mutex{}}
+	e.ErrorFormat = NewErrorFormatter().multiErrFormatter
+	return e
+}
 
-	switch {
-	case l == 0:
-		out = e.msg
-	case l == 1:
```

Comment:
Diff vyzera trocha divoko, no v principe sa kod presunul do `ErrorFormatter` a pridala sa moznost nastavit vlastny callback. Asi to nie je potrebne nejak extra prechadzat, kedze chyby su pokryte testami v celej apke.